In [1]:
from astroquery.eso import Eso
from astropy.coordinates import SkyCoord

In [2]:
from astropy.coordinates import SkyCoord # import the SkyCoord class from the astropy.coordinates module
import astropy.units as u # import the astropy.units module

coords = SkyCoord.from_name('Sgr A*') # create a SkyCoord object from the name of the source
radius = 60 *u.arcmin # set the radius of the search to 20 arcminutes

In [3]:
eso = Eso() # create an instance of the ESO class 
eso.ROW_LIMIT = None # Does this limit to 1000?
# eso.ROW_LIMIT = 2000 # Doesn't seem to increase?

In [4]:
collections = eso.list_collections() # list all collections
collections

['081.C-0827',
 '092.A-0472',
 '096.B-0054',
 '1100.A-0528',
 '1101.A-0127',
 '193.D-0232',
 '195.B-0283',
 '196.B-0578',
 '196.D-0214',
 '197.A-0384',
 '198.A-0708',
 '60.A-9284H',
 '60.A-9493',
 'ADHOC',
 'ALCOHOLS',
 'ALLSMOG',
 'ALMA',
 'AMAZE',
 'AMBRE',
 'APEX-SciOps',
 'ATLASGAL',
 'ePESSTOplus',
 'ESPRESSO',
 'ESSENCE',
 'FDS',
 'FEROS',
 'Fornax3D',
 'FORS2-SPEC',
 'GAIAESO',
 'GCAV',
 'GIRAFFE',
 'GOODS_FORS2',
 'GOODS_ISAAC',
 'GOODS_VIMOS_IMAG',
 'GOODS_VIMOS_SPEC',
 'GW170817',
 'HARPS',
 'HAWKI',
 'HUGS',
 'INSPIRE',
 'KIDS',
 'KMOS',
 'LEGA-C',
 'LESS',
 'MUSE',
 'MUSE-DEEP',
 'MUSE-STD',
 'MW-BULGE-PSFPHOT',
 'NGTS',
 'PESSTO',
 'PHANGS',
 'PIONIER',
 'SPHERE',
 'SUPER',
 'UltraVISTA',
 'UVES',
 'UVES_SQUAD',
 'VANDELS',
 'VEGAS',
 'VEILS',
 'VEXAS',
 'VHS',
 'VIDEO',
 'VIKING',
 'VIMOS',
 'VINROUGE',
 'VIPERS',
 'VISIONS',
 'VMC',
 'VPHASplus',
 'VST-ATLAS',
 'VVV',
 'VVVX',
 'XQ-100',
 'XSGRB',
 'XSHOOTER',
 'XSL',
 'ZCOSMOS']

In [5]:
instruments = eso.list_instruments() # list all collections
instruments

['amber',
 'apex',
 'crires',
 'eris',
 'espresso',
 'fiat',
 'fors1',
 'fors2',
 'giraffe',
 'gravity',
 'hawki',
 'isaac',
 'kmos',
 'matisse',
 'midi',
 'muse',
 'naco',
 'omegacam',
 'pionier',
 'sinfoni',
 'sphere',
 'uves',
 'vimos',
 'vircam',
 'visir',
 'wlgsu',
 'xshooter']

**Define some search params**

In [6]:
instrument = 'hawki'       # set the instrument to HAWKI
pi_coi = 'SCHOEDEL/ DO/ ECKART/ GIRARD/ MUZIC/ NEUMAYER/ NISHIYAMA/ SANCHEZ BERMUDEZ' # set the Co-Investigators
ra = coords.ra             # get the right ascension of the source
dec = coords.dec           # get the declination of the source
radius = radius.to('deg')  # convert the radius to degrees

**Could we make all of these consistent?**   
**Do we always have to give instrument?**

```python
table_collections   = eso.query_collections(instrument, ra=ra.value, dec=dec.value, radius=radius.value) 
table_instrument    = eso.query_instrument(instrument, ra=ra.value, dec=dec.value, radius=radius.value) 
table_main          = eso.query_main(instrument, ra=ra.value, dec=dec.value, radius=radius.value)
```

In [7]:
table_collections   = eso.query_collections(instrument, ra=ra.value, dec=dec.value, radius=radius.value) 
table_instrument    = eso.query_instrument(instrument, ra=ra.value, dec=dec.value, radius=radius.value) 
table_main          = eso.query_main(column_filters={'instrument': instrument}, ra=ra.value, dec=dec.value, radius=radius.value)

/opt/anaconda3/envs/tst/lib/python3.9/site-packages/pyvo/dal/query.py:339: DALOverflowWarning: Partial result set. Potential causes MAXREC, async storage space, etc.
  warn("Partial result set. Potential causes MAXREC, async storage space, etc.",


Setting instrument as keyword does not work! 

In [11]:
table_collections   = eso.query_collections(instrument, ra=ra.value, dec=dec.value, radius=radius.value) 
table_collections   = eso.query_collections(instrument=instrument, ra=ra.value, dec=dec.value, radius=radius.value)

TypeError: 'NoneType' object is not iterable

**Note error limiting to 1000 results - can we fix this? Or make them for a async?**  
**Do we offer async yet?**

/opt/anaconda3/envs/tst/lib/python3.9/site-packages/pyvo/dal/query.py:339: DALOverflowWarning: Partial result set. Potential causes MAXREC, async storage space, etc.
  warn("Partial result set. Potential causes MAXREC, async storage space, etc.",

In [8]:
print(f'Collentions - length {len(table_collections)} - columns {len(table_collections.colnames)}')
print(f'Instrument - length {len(table_instrument)} - columns {len(table_collections.colnames)}')
print(f'Main - length {len(table_main)} - columns {len(table_collections.colnames)}')

Collentions - length 136 - columns 54
Instrument - length 1000 - columns 54
Main - length 1000 - columns 54


None of them **need** ``ra, dec, radius``... Good... 

In [9]:
table_collections   = eso.query_collections(instrument) 
# table_instrument    = eso.query_instrument(instrument, column_filters={'pi_coi':pi_coi})
# table_main          = eso.query_main(column_filters={'instrument': instrument, 'pi_coi':pi_coi})